### PR TITLE
Make root oc cmd dynamic in all command output

### PR DESCRIPTION
**Pull request by juanvallejo**
Related BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1256274

Suggestions to use the "oc" command are hardcoded into some commands (and their help output). This updates the help output and other instances in those commands to be dynamic.

###### Before
```
$ openshift cli login -u test -p test
You have access to the following projects and can switch between them with 'oc project <projectname>':
```
###### After
```
$ openshift cli login -u test -p test
You have access to the following projects and can switch between them with 'openshift cli project <projectname>':
```

### CODE REVIEW

**Comment by juanvallejo** [comment]
[test]

**Comment by juanvallejo** [comment]
re[test]

**Comment by juanvallejo** [comment]
re[test]

File: docs/man/man1/oc-status.1 [hunk 1]
```diff
@@ -18,8 +18,8 @@ Show a high level overview of the current project
 .PP
 This command will show services, deployment configs, build configurations, and active deployments.
 If you have any misconfigured components information about them will be shown. For more information
-about individual items, use the describe command (e.g. oc describe buildConfig,
-oc describe deploymentConfig, oc describe service).
+about individual items, use the describe command (e.g. oc  describe buildConfig,
+oc  describe deploymentConfig, oc  describe service).
```
Comment (fabianofranz) [inline docs/man/man1/oc-status.1]: It seems to be printing two spaces after `oc`.

File: docs/man/man1/openshift-cli-status.1 [hunk 2]
```diff
@@ -18,8 +18,8 @@ Show a high level overview of the current project
 .PP
 This command will show services, deployment configs, build configurations, and active deployments.
 If you have any misconfigured components information about them will be shown. For more information
-about individual items, use the describe command (e.g. oc describe buildConfig,
-oc describe deploymentConfig, oc describe service).
+about individual items, use the describe command (e.g. openshift cli  describe buildConfig,
+openshift cli  describe deploymentConfig, openshift cli  describe service).
```
Comment (fabianofranz) [inline docs/man/man1/openshift-cli-status.1]: Same here.

File: pkg/cmd/cli/cmd/login.go [hunk 3]
```diff
@@ -180,6 +182,10 @@ func (o LoginOptions) Validate(args []string, serverFlag string) error {
 
 // RunLogin contains all the necessary functionality for the OpenShift cli login command
 func RunLogin(cmd *cobra.Command, options *LoginOptions) error {
+	if options.CommandName == "" {
```
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/login.go]: Looks like this check belongs to `Complete`.

File: pkg/cmd/cli/cmd/status.go [hunk 4]
```diff
@@ -59,11 +60,15 @@ type StatusOptions struct {
 // NewCmdStatus implements the OpenShift cli status command.
 func NewCmdStatus(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
 	opts := &StatusOptions{}
+	cmdName := fullName
+	if strings.HasSuffix(fullName, name) {
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/status.go]: this is concerning... we shouldn't have to mess with this value...

Comment (juanvallejo) [inline pkg/cmd/cli/cmd/status.go]: @liggitt Without this, the help output for this command would include `status` in cases where it shouldn't:

`openshift cli status -h` 
```
...
This command will show services, deployment configs, build configurations, and active deployments.
If you have any misconfigured components information about them will be shown. For more information
about individual items, use the describe command (e.g. openshift cli status describe buildConfig,
openshift cli status describe deploymentConfig, openshift cli status describe service).
```

Comment (liggitt) [inline pkg/cmd/cli/cmd/status.go]: then that's not the right value to use. We should not assume that stripping the last segment off the command will give us the root cli command name

Comment (fabianofranz) [inline pkg/cmd/cli/cmd/status.go]: Yeah, `NewCmdStatus` should take an additional argument with what you want here (`oc|openshift cli|...`), ready to use. I *think* `fullName` gives you that (I mean `fullName` in `cli.go`, the one you have before appending `StatusRecommendedName`).

**Comment by juanvallejo** [comment]
@liggitt @fabianofranz Thanks for the feedback! I went ahead and passed an extra argument to `NewCmdStatus` from `cli.go` to display help output properly.

**Comment by fabianofranz** [comment]
LGTM, @liggitt anything else?

File: pkg/cmd/cli/cmd/status.go [hunk 5]
```diff
@@ -57,13 +57,14 @@ type StatusOptions struct {
 }
 
 // NewCmdStatus implements the OpenShift cli status command.
-func NewCmdStatus(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+// helpName is the path from root cmd to the parent of this cmd.
+func NewCmdStatus(name, helpName, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/status.go]: `helpName` is not clear... `baseCLIName`?

**Comment by juanvallejo** [comment]
@liggitt 
> helpName is not clear... baseCLICommand?

Sounds good, [also added a test for the help output](https://github.com/openshift/origin/pull/9985/commits/e94481737949c612463e53d87119eed6c71e8f95#diff-d53f7b5afddc55ed899ea1244ff7985cR73)

File: test/cmd/basicresources.sh [hunk 6]
```diff
@@ -70,6 +70,12 @@ echo "version reporting: ok"
 os::test::junit::declare_suite_end
 
 os::test::junit::declare_suite_start "cmd/basicresources/explain"
```
Comment (liggitt) [inline test/cmd/basicresources.sh]: needs a unique suite name

**Comment by juanvallejo** [comment]
@liggitt 
> needs a unique suite name

Thanks for catching this

**Comment by liggitt** [comment]
LGTM

**Comment by fabianofranz** [comment]
@juanvallejo good to go?

**Comment by juanvallejo** [comment]
@fabianofranz Yeah, everything looks good to me

**Comment by fabianofranz** [comment]
[merge]

**Pull request merged**